corehq/apps/geospatial/static/geospatial/js/geo_config.js

```diff
@@ -46,6 +50,33 @@ hqDefine("geospatial/js/geo_config", [
             return self.selectedGroupMethod() === minMaxGroupingName;
         });
 
+        self.selectedAlgorithm = ko.observable();
+        self.apiToken = ko.observable(data.api_token);
+
+        self.captureApiToken = ko.computed(function () {
+            return self.selectedAlgorithm() === ROAD_NETWORK_ALGORITHM;
+        });
+
+        self.validateApiToken = function () {
```

Comment:
I know for connection settings the user can choose to test the connection before saving it, but personally I don't think we should follow that approach here. Reason being:
- The validity of the connection settings can change anytime from `valid` to `invalid` to `valid` again.
- The token's validity cannot change from invalid to valid, but can change valid to invalid

I suggest we test the token's validity when the user clicks save. If the token is valid, we continue, otherwise we tell the user it's not valid.

Reply:
> I suggest we test the token's validity when the user clicks save. If the token is valid, we continue, otherwise we tell the user it's not valid.

That does sound good.

Just one place we might block the user is, if they wish to change the token but the mapbox api is down and so the validation is unable to run.
I assume we will only run the test if the value has changed and not otherwise. That logic will need to be added as well.

So, there are pros and cons for both. Though I'd like if users would test the token if they change it.

Reply:
I figure we'd know when Mapbox is down from the status we get back. In that case we 1. tell the user and 2. proceed and fall back on the logic that tells the user about the invalid token when they run the algorithm.

Reply:
Right.
I was just comparing the benefits with the amount of code this code add. Feel free to give it a go and see how you feel about it. I do see an advantage of verifying the token before its saved, manually by the user or automatically by HQ itself.

Reply:
I initially thought of making the request on save to check validity, but then again I don't think we should "force" a sample check. We don't do that for connection settings so I think it's better that we don't do it for this also. 

The button technically don't have to exist and is purely for the user to make sure they copied the right value. If they are sure and don't see the need to test it that's up to them. 

If we make a sample request on save (even if it's only if the token value changed) that means there's technically a side-effect eating through their free budget on Mapbox; what if the users don't want that?

Reply:
>  their free budget on Mapbox; what if the users don't want that?

I doubt one request is going to hurt the number of free requests they have or even get noticed. So, we would have been fine, if we were to do it.

Reply:
> I doubt one request is going to hurt the number of free requests they have or even get noticed

I do agree, but for me it's a principle thing; I don't think there should be background side-effects on a seemingly unrelated page like "Settings". The user expects the token to be used only when it's actually necessary (running disbursement), unless explicitly actioned elsewhere.